### DISCUSSION TITLE
Dropdown sorting help

### DISCUSSION THREAD

**Discussion by ericberringer** [nss-day-cohort-46/How-to-Ask-for-Help#27] [Ask For Help] [answered]
#### Have you searched the other issue tickets? If not, do that first. 
- [ x] Yes
- If No, stop and search.

####  REQUIRED Add Tags to Issue Ticket and Assign to Project Board
 

- [ x] Added label to specific project/assignment that you need help with
- [x ] Added to `Student-Help-Tracking` project board

#### REQUIRED Describe your issue:
Trying to figure out how to filter the products based on their category.

#### REQUIRED List any error messages that appear:
None

#### REQUIRED What have you googled? 

None, because I didn't follow directions 😈 

#### REQUIRED: What have you tried? Tell us everything you have tried. 
Referencing other projects.

#### REQUIRED: Copy/paste link TO THE FILE on your branch that you are having issues with

**Your link:**

https://github.com/nss-day-cohort-46/sprinkles-of-joy-jawa-juice-cakes/blob/EB-Sprinkles-DontKnow/scripts/products/ProductList.js

#### Code Snippet

```
const render = () => {
  contentTarget.innerHTML = bakeryProducts.map(product => {
    const productCategory = bakeryCategories.find(category => category.id === product.categoryId)

    return Product(product, productCategory)
  }).join("")
}

eventHub.addEventListener("categorySelected", event => {
  const catId = parseInt(event.detail.selectedCategory)
  bakeryProducts = useProducts()
  bakeryCategories = useCategories()

  const matchingCategories = bakeryCategories.find(category => category.id === catId)
  const matchingProducts = bakeryProducts.filter(product => product.categoryId === catId)

})
```

**Comment by ericberringer** [comment, marked as answer]
This fixes the dropdown sorting issue. 
- Added conditional to our listening event to sort products by category or all categories.
- Removed parseInt from parseInt(event.detail.selectedCategory) and added to catId within our filter method on useProducts().
- Scort showed us a cool unary operator "+" that means parseInt, but he said don't use it EVER!
- Learned you can chain a .filter() directly to useProducts().
- Passed in render() after each item in our conditional to print the content we needed per each condition.

ProductList.js
"categorySelected" is our Custom Event being heard from CategorySelect.js
```
eventHub.addEventListener("categorySelected", event => {
  const catId = event.detail.selectedCategory
  if(catId !== "0") {
    bakeryProducts = useProducts().filter(product => product.categoryId === +catId)
    // + means parseInt, but don't do it, Scort's orders.
    render()
  } else {
    bakeryProducts = useProducts()
    render()
  }
  
})
```

**Comment by nicoletatum** [comment]
We had the same problem and this worked!! Thanks

**Comment by SubtleCo** [comment]
Oh cool! @Scott47 can you shed a little light on why not to use + as a shorthand parseInt?

**Reply by Scott47** [reply]
@SubtleCo Unary operators are super convenient and I really like them. The only drawback I see is that they are abstractions that make your code less readable. In this case, `parseInt` is much more explicit and readable than `+`.  Having said that, you are introduced to them often as a beginner. The unary operator for incrementing is `++`,  which we use in pre-work and in the jumpstart curriculum with for loops (i.e. `for(let i=0; i <= array.length; i++)`).  Here is a handy link: [unary operators](https://www.javascripttutorial.net/javascript-unary-operators/)
